### ISSUE TITLE
Looks like local variables are unnecessarily made unique by postfixing "_"

### ISSUE THREAD

**Issue by sjaakd** [mapstruct/mapstruct#524]

**Comment by gunnarmorling** [comment]
Is it a specific case where this is happing?

**Comment by gunnarmorling** [comment]
Hey @sjaakd, could you add an example where this is happening?

**Comment by sjaakd** [comment]
> Hey @sjaakd, could you add an example where this is happening?

I'll provide an example.. Need to look into one of my mappers @work.

**Comment by sjaakd** [comment]
Example: run `org.mapstruct.ap.test.collection.CollectionMappingTest`

The generated code looks like:

```java
   @Override
    public Target sourceToTarget(Source source) {
        if ( source == null ) {
            return null;
        }

        Target target = new Target();

        if ( source.getStringSet2() != null ) {
            if ( target.getStringListNoSetter2() != null ) {
                Collection<String> targetCollection = source.getStringSet2();
                if ( targetCollection != null ) {
                    target.getStringListNoSetter2().addAll( targetCollection );
                }
            }
        }
        if ( source.getIntegerSet() != null ) {
            target.setSet( new HashSet( source.getIntegerSet() ) );
        }
        if ( source.getStringList2() != null ) {
            if ( target.getStringListNoSetter() != null ) {
                Collection<String> targetCollection_ = source.getStringList2();
                if ( targetCollection_ != null ) {
                    target.getStringListNoSetter().addAll( targetCollection_ );
                }
            }
        }
```

`targetCollection` is made unique in the scope of the next method by prefixing it with an underscore (as `targetCollection_`). If you've got many `Collections` as target, each one will get an additional underscore. Note: this is not needed, because `targetCollection` lives in its own scope. Problems will arise only when the method signature of `public Target sourceToTarget(Source source) ` will have a parameter named `targetCollection`, like this: `public Target sourceToTarget(Source targetCollection) ` which is a bit strange here, but not with update methods (`@MappingTarget`).

To be honest, I think that fixing this will require another approach to the mechanism we use now. So I'm not sure whether its worthwhile to invest in that.

Shall I close this one?

**Comment by gunnarmorling** [comment]
I agree, there is no urgent need to fix it. I will leave it to your judgement whether to close it or leave it open for later fixing. Indeed we'd need more intelligent means of making names unique, with an understanding of blocks/scopes within generated methods. All in all I don't feel it's something we should put much effort in.

**Comment by sjaakd** [comment]
> All in all I don't feel it's something we should put much effort in.

agreed. I'll keep it open, just to remember and mark it as improvement. Pity though that we can't apply some priority :smile:

**Comment by filiphr** [comment]
I think that this one was resolved via #941
